pkg/block/indexheader/reader_pool.go

```diff
@@ -113,6 +115,21 @@ func NewReaderPool(logger log.Logger, lazyReaderEnabled bool, lazyReaderIdleTime
 				}
 			}
 		}()
+
+		if p.lazyReaderIdleDeleteTimeout > 0 {
+			deleteCheckFreq := p.lazyReaderIdleDeleteTimeout / 10
+
+			go func() {
+				for {
+					select {
+					case <-p.close:
+						return
+					case <-time.After(deleteCheckFreq):
+						p.deleteIdleReaders()
```

Comment:
I image with more support of `lazyDownloadFunc` in the future, some index headers are always eagerly downloaded and some can be lazily downloaded.
For deleting idle index headers, I am hoping to only clean up those unused lazily downloadable index headers only and keep the rest on disk even though they never get accessed. For query latency purpose.